test/macros.jl

```diff
@@ -176,6 +182,29 @@ function build_constraint_keyword_test(ModelType::Type{<:JuMP.AbstractModel})
     end
 end
 
+struct MyConstrType end
+struct BadPosArg end
+function JuMP.build_constraint(
+    _error::Function,
+    f,
+    set,
+    extra::Type{MyConstrType};
+    d = false,
+)
+    return JuMP.build_constraint(_error, f, set)
+end
+function build_constraint_extra_arg_test(ModelType::Type{<:JuMP.AbstractModel})
+    @testset "build_constraint with extra positional arguments" begin
+        model = ModelType()
+        @variable(model, x)
+        cref = @constraint(model, x == 0, MyConstrType)
+        @test JuMP.constraint_object(cref).set isa MOI.EqualTo{Float64}
```

Comment:
I wonder if you could change the test to do something like modify the set so it isn't `MOI.EqualTo(0.0)`. This doesn't test that the overload worked, only that it didn't error.

Reply:
Ok, I modified this test to be more involved to verify it is working as expected.